### PR TITLE
HDDS-7290. provide a config to increase the list batch size in OzoneFileSystem

### PR DESCRIPTION
**Pull request by devmadhuu**
## What changes were proposed in this pull request?

Currently the value of list page size for listing items of ozone fs sub-commands output is set as hardcoded value 1024. 

A new configuration "ozone.client.fs.default.listing.page.size" will be provided to configure the value.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7290

## How was this patch tested?

Manual testing and configured value is reflecting.
image
image

### CODE REVIEW

**Review by adoroszlai — COMMENTED** [review]
Thanks @devmadhuu for working on this.

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 1]
```diff
@@ -3358,5 +3358,22 @@
...
+    <name>ozone.client.fs.default.listing.page.size</name>
+    <value>1024</value>
+    <tag>OZONE, CLIENT</tag>
+    <description>
+      Default listing page size value used for listing items on fs related sub-commands.
+    </description>
+  </property>
+
+  <property>
+    <name>ozone.client.fs.max.listing.page.size</name>
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-hdds/common/src/main/resources/ozone-default.xml:3371]: @nandakumar131 , pls suggest on this comment. I already had discussion with @adoroszlai on this comment.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java [hunk 2]
```diff
@@ -91,6 +94,8 @@ public class BasicOzoneFileSystem extends FileSystem {
   static final Logger LOG =
       LoggerFactory.getLogger(BasicOzoneFileSystem.class);
 
+  private static int listingPageSize = 1024;
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:97]: yes, resolved.

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/OMConfigKeys.java [hunk 3]
```diff
@@ -405,4 +405,17 @@ private OMConfigKeys() {
...
+
+  // Items listing page size for fs client sub-commands output
+  public static final String OZONE_CLIENT_FS_MAX_LISTING_PAGE_SIZE =
+      "ozone.client.fs.max.listing.page.size";
+
+  public static final String
+      OZONE_CLIENT_FS_LISTING_PAGE_SIZE =
+      "ozone.client.fs.default.listing.page.size";
+  public static final int
+      OZONE_CLIENT_FS_LISTING_PAGE_SIZE_DEFAULT = 1024;
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/OMConfigKeys.java:417]: Ok

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 4]
```diff
@@ -3358,5 +3358,22 @@
     </description>
   </property>
 
+  <property>
+    <name>ozone.client.fs.default.listing.page.size</name>
+    <value>1024</value>
+    <tag>OZONE, CLIENT</tag>
+    <description>
+      Default listing page size value used for listing items on fs related sub-commands.
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-hdds/common/src/main/resources/ozone-default.xml]: @adoroszlai , pls re-review. Comment is fixed.

File: hadoop-hdds/common/src/main/resources/ozone-default.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-hdds/common/src/main/resources/ozone-default.xml]: @adoroszlai , pls re-review, comment is fixed.

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/OMConfigKeys.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/OMConfigKeys.java]: @adoroszlai , pls re-review, comment is fixed.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java [hunk 5]
```diff
@@ -92,6 +94,9 @@ public class BasicRootedOzoneFileSystem extends FileSystem {
   static final Logger LOG =
       LoggerFactory.getLogger(BasicRootedOzoneFileSystem.class);
 
+  private static int listingPageSize =
+      OZONE_CLIENT_FS_LISTING_PAGE_SIZE_DEFAULT;
```
Comment (adoroszlai) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java:98]: Please change this to not be `static`, as findbugs complains about it.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java [hunk 6]
```diff
@@ -91,6 +93,9 @@ public class BasicOzoneFileSystem extends FileSystem {
   static final Logger LOG =
       LoggerFactory.getLogger(BasicOzoneFileSystem.class);
 
+  private static int listingPageSize =
+      OZONE_CLIENT_FS_LISTING_PAGE_SIZE_DEFAULT;
```
Comment (adoroszlai) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:97]: Please change this to not be `static`, as findbugs complains about it.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java (hunk 5 above)
Comment (devmadhuu) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java]: Thanks @adoroszlai for reviewing this PR. I'll run all checks locally now before pushing. I didn't aware of this earlier. Thanks for guiding me. Please re-review. Its fixed and passing all checks

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java (hunk 6 above)
Comment (devmadhuu) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java]: Thanks @adoroszlai for reviewing this PR. I'll run all checks locally now before pushing. I didn't aware of this earlier. Thanks for guiding me. Please re-review. Its fixed and passing all checks

**Review by adoroszlai — COMMENTED** [review]
Thanks @devmadhuu for updating the patch.  I just noticed that page size configured over max. allowed affects all FS operations (e.g. `-put`), not just list-related ones.  Is that OK?

**Comment by devmadhuu** [comment]
> 

@adoroszlai , valid point, however before doing this change I noticed that same hard coded variable value was being used for all fs operations , I just made it configurable that variable. Pls let me know if I am missing something.

**Comment by adoroszlai** [comment]
> owever before doing this change I noticed that same hard coded variable value was being used for all fs operations , I just made it configurable that variable. Pls let me know if I am missing something.

Previously there was no precondition check, since the value was hard-coded.  Adding a check this way affects all operations, even those that to which page size does not apply.

Instead of throwing exception if the page size config exceeds max. allowed, how about limiting it (and maybe logging a warning)?

**Pull request closed without merging**